### PR TITLE
Fix image block render in Xcode 12

### LINKED ISSUE #25431: Images fail to load when building Gutenberg using Xcode 12
**Describe the bug**

When building the demo app on Xcode 12, images can no longer be added to the Image block.
It looks to affect any block that supports images (videos appear unaffected).

**To reproduce**
1. Using Xcode 12, build and run the app
2. Add an Image block
3. Add an image to that block (its irrelevant whether or not the app has permission to just some or all photos)
4. Notice the image appears to load (loading bar advances) but the image fails to appear in the block

**Expected behavior**
Building the demo app on Xcode 12 shouldn't cause images to fail to load.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone 11
 - OS: iOS 14
 - Demo App Version https://github.com/WordPress/gutenberg/commit/3ed0910f976a486b4a73a700fed1d213e3dac5b8

### PR DESCRIPTION
**Pull request by guarani**
## Description

This PR fixes #25431 which is a bug that prevents Image blocks from rendering their images. This only affects builds made using Xcode 12.

The bug itself is in React Native itself and was fixed in RN 0.63. Since we're on RN 0.61.5, we need to apply a patch (as suggested here https://github.com/facebook/react-native/issues/29279#issuecomment-657201709).

The patch was applied using the following steps found here: https://github.com/facebook/react-native/issues/29279#issuecomment-657201709

## How has this been tested?

Ensure Xcode 12 is installed

1. Run the packager: npm run native start
2. Run the iOS demo app (open `packages/react-native-editor/ios/gutenberg.xcworkspace` and run from Xcode 12 *
3. Verify that the image block in the demo content renders its image correctly (that's to say, the image should be visible)

*There appears to be an issue running npm run native ios with Xcode 12

## Screenshots 

Screen Shot 2020-09-18 at 17 40 41

## Types of changes
- Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: patches/react-native+0.61.5.patch [hunk 1]
```diff
@@ -53,3 +67,10 @@ index 40c0cda..646f137 100644
    }
    return ancestor == shadowView;
  }
+diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
```
Comment (ceyhun) [inline patches/react-native+0.61.5.patch:70]: This change seems like it's not related, will we need this as well?

File: patches/react-native+0.61.5.patch [hunk 2]
```diff
@@ -21,6 +21,20 @@ index 65fa2bb..c5f265b 100644
      index++;
    }
  
+diff --git a/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m b/node_modules/react-native/Libraries/Image/RCTUIImageViewAnimated.m
```
Comment (ceyhun) [inline patches/react-native+0.61.5.patch:24]: We'll need to duplicate this patch for WPiOS as well. You can check here to see how it's done for other patches: https://github.com/wordpress-mobile/WordPress-iOS/blob/e221f2345d1e953737458ed1559621df030683af/Podfile#L223

**Review by ceyhun — COMMENTED** [review]
Tested both with Xcode 11, iOS 13 sim and Xcode 12, iOS 14 sim locally and worked great!

File: patches/react-native+0.61.5.patch (hunk 1 above)
Comment (guarani) [reply patches/react-native+0.61.5.patch]: Thanks for catching that!

File: patches/react-native+0.61.5.patch (hunk 2 above)
Comment (guarani) [reply patches/react-native+0.61.5.patch:24]: True! Thanks for the pointer 👍

**Comment by guarani** [comment]
Thanks for reviewing @ceyhun! I've made the updates and created a matching PR to apply this patch to the WPiOS repo: https://github.com/wordpress-mobile/WordPress-iOS/pull/14988

If you could take another look at this and the new PR that would be great!

**Review by ceyhun — APPROVED** [review]
Tested again with Xcode 11 & 12 and works great, thanks!

**Comment by guarani** [comment]
Thank you very much @ceyhun, much appreciated!
